### PR TITLE
DOC: add example of pure_callback with custom_jvp

### PR DESCRIPTION
**Pull request by jakevdp**
This is an example that is meant to come at the end of a more general discussion of `pure_callback`, `debug.callback`, and other callback mechanisms. This sort of question has come up frequently enough recently that I thought it would be useful to have a stub file defined into which we can fill-in details later.

Preview here: https://jax--13587.org.readthedocs.build/en/13587/notebooks/external_callbacks.html

### CODE REVIEW

File: docs/notebooks/external_callbacks.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+This is a work-in-progress, and will be updated soon.
+
+*TODO(jakevdp, sharadmv): fill-in some simple examples of {func}`jax.pure_callback`, {func}`jax.debug.callback`, {func}`jax.debug.print`, and others.*
+
++++ {"id": "dF7hoWGQUneJ"}
+
+## Example: `pure_callback` with `custom_jvp`
+
+One powerful way to take advantage of pure callbacks is to combine them with `custom_jvp` (see [Custom derivative rules](https://jax.readthedocs.io/en/latest/notebooks/Custom_derivative_rules_for_Python_code.html) for more details on `custom_jvp`). Suppose we want to create a JAX-compatible wrapper for a scipy or numpy function that is not yet available in the `jax.scipy` or `jax.numpy` wrappers.
```
Comment (sharadmv) [inline docs/notebooks/external_callbacks.md:30]: I think it might be worth linking the pure_callback docs here (https://jax.readthedocs.io/en/latest/_autosummary/jax.pure_callback.html)

File: docs/notebooks/external_callbacks.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+:id: q3qh_4DrWxdQ
+:outputId: c46b0bfa-96f3-4629-b9af-a4d4f3ccb870
+:tags: [raises-exception]
+
+jax.grad(j1)(z)
+```
+
++++ {"id": "PtYeJ_xUW09v"}
+
+Let's define a custom gradient rule for this. Looking at the definition of the [Bessel Function of the First Kind](https://en.wikipedia.org/?title=Bessel_function_of_the_first_kind), we discover that there is a relatively straightforward recurrance relationship for the derivative with respect ot the argument `z`:
```
Comment (sharadmv) [inline docs/notebooks/external_callbacks.md:106]: Nit: "ot" -> "to"

Comment (sharadmv) [inline docs/notebooks/external_callbacks.md:106]: Nit: "recurrance" -> "recurrence"

**Review by sharadmv — APPROVED** [review]
Thanks Jake, this is great!

File: docs/notebooks/external_callbacks.md (hunk 1 above)
Comment (jakevdp) [reply docs/notebooks/external_callbacks.md]: Done

File: docs/notebooks/external_callbacks.md (hunk 2 above)
Comment (jakevdp) [reply docs/notebooks/external_callbacks.md]: Done

Comment (jakevdp) [reply docs/notebooks/external_callbacks.md]: Done
